compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -1949,15 +1948,27 @@ private STNode parseParameterRhs(SyntaxKind tokenKind) {
      */
     private STNode parseParameter(SyntaxKind prevParamKind, boolean isParamNameOptional) {
         STNode annots;
+        STNode inclusionSymbol;
         STToken nextToken = peek();
         switch (nextToken.kind) {
             case AT_TOKEN:
                 annots = parseOptionalAnnotations();
+                if (peek().kind == SyntaxKind.ASTERISK_TOKEN) {
+                    inclusionSymbol = parseAsteriskToken();
+                } else {
+                    inclusionSymbol = STNodeFactory.createEmptyNode();
+                }
```

Comment:
This will not handle syntax errors that could come after annotations. We need to introduce a another level to parse the content after the annots.

Reply:
+1 I think we need to handle recovery for the included record param

Reply:
isn't it enough going to recovery inside the parseTypeDescriptor when there's an error followed by annotations? 
Reason is error handler will never pick `INSERT *` as the optimal path. It will always go for the required-param, defaultable-param or rest-param

Reply:
sorry. we can something like `@untainted invalid token *Options options`
+1

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/26882